src/Tvheadend.cpp

```diff
@@ -367,6 +367,34 @@ PVR_ERROR CTvheadend::GetChannels(bool radio, kodi::addon::PVRChannelsResultSet&
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR CTvheadend::GetChannelStreamProperties(const kodi::addon::PVRChannel& channel, std::vector<kodi::addon::PVRStreamProperty>& properties)
+{
+  CLockObject lock(m_mutex);
+
+  if (!Settings::GetInstance().GetStreamingHTTP())
+    return PVR_ERROR_NO_ERROR;
+
+  auto it = m_channels.find(channel.GetUniqueId());
+  if (it == m_channels.end())
+    return PVR_ERROR_FAILED;
+
+  std::string url = "http://";
+  url += Settings::GetInstance().GetUsername() + ":" + Settings::GetInstance().GetPassword() + "@";
```

Comment:
please handle empty username, password correctly.

Reply:
if username is empty, we won't get to this stage

Reply:
why?

Reply:
how would the addon connect to the backend without valid username / pw?

Reply:
In fact you are right. Currently it is the case that kodi calls 'GetChannelStreamProperties' only after successful backend connection was established by the addon.  But there is no guarantee for this. Think a bout async connect. May I suggest to make the code a bit more robust?

Reply:
if username or password are empty or wrong, the channel switch will fail later when opening the url.  this is the desired behavior.

Reply:
@FernetMenta it's possible to configure tvheadend to accept no or any username/password, so you can't AFAIK rely on execution not reaching this place if either username or password is empty.

Not sure what the most robust solution would be here. Maybe omit the credentials part completely if either username or password is empty.